lang/cs/Org.Apache.REEF.Common/Telemetry/MetricsService.cs

```diff
@@ -65,13 +89,64 @@ public void OnNext(IContextMessage contextMessage)
                     //// For the counters from multiple evaluators with the same counter name, the value should be aggregated here
                     //// We also need to consider failure cases.  
                     _counters[counter.Name] = counter;
+
+                    if (_totalIncrementSinceLastSink.ContainsKey(counter.Name))
+                    {
+                        _totalIncrementSinceLastSink[counter.Name] += counter.Value - c.Value;
+                    }
+                    else
+                    {
+                        _totalIncrementSinceLastSink.Add(counter.Name, counter.Value - c.Value);
+                    }
```

Comment:
Would it make sense to hide this logic inside some abstraction, like CounterData, which would keep reference to a counter and lastIncrement value?
them MetricsService wouldn't need to maintain 2 dictionaries